### PR TITLE
Cycle and Include tags

### PR DESCRIPTION
**Pull request by Spadavecchia**

### CODE REVIEW

File: lib/liquid/nimble_parser.ex [hunk 1]
```diff
@@ -146,14 +147,206 @@ defmodule Liquid.NimbleParser do
...
+    |> parsec(:ignore_whitespaces)
+    |> optional(parsec(:var_assignation))
+
+  defparsecp(:var_assignation, var_assignation)
+
+  # {% include 'color' with 'red' %}
+  with_param =
+    empty()
+    |> ignore(string("with"))
+    |> concat(parsec(:ignore_whitespaces))
```
[the remark this answers is not in the document]
Comment (Spadavecchia) [reply lib/liquid/nimble_parser.ex]: Sure, we refactored a lot of code when discovered the trick to split the code in multiple modules

**Comment by Spadavecchia** [comment]
We have add lot of tests in the current WIP

File: lib/liquid/combinators/general.ex [hunk 2]
```diff
@@ -7,7 +7,10 @@ defmodule Liquid.Combinators.General do
   # Codepoints
   @horizontal_tab 0x0009
   @space 0x0020
+  @colon 0x003A
   @point 0x002E
+  @comma 0x002C
+  @apostrophe 0x0027
```
[the remark this answers is not in the document]
Comment (Eptis) [reply lib/liquid/combinators/general.ex]: Can you move the relevant ones into this PR?

[the remark this answers is not in the document]
Comment (Spadavecchia) [reply lib/liquid/combinators/general.ex]: Yes, I put all relevant in this PR

File: lib/liquid/combinators/tags/raw.ex [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  def not_close_tag_raw do
+    empty()
+    |> ignore(utf8_char([]))
+    |> parsec(:raw_content)
+  end
+
+  def raw_content do
+    empty()
+    |> repeat_until(utf8_char([]), [
+      string(General.codepoints().start_tag)
```
[the remark this answers is not in the document]
Comment (Spadavecchia) [reply lib/liquid/combinators/tags/raw.ex]: Yes, indeed I test it and generate an error. We need test it with more edge cases.

**Comment by Spadavecchia** [comment]
PR too big, splitting it into small PRs
